finetune/target_models/masked_language_model.py

```diff
@@ -0,0 +1,149 @@
+import tensorflow as tf
+from tensorflow import TensorShape as TS
+import numpy as np
+
+from finetune.errors import FinetuneError
+from finetune.base import PredictMode, BaseModel
+from finetune.base_models import RoBERTa, BERT
+from finetune.input_pipeline import BasePipeline
+from finetune.nn.target_blocks import masked_language_model
+
+
+class MaskedLanguageModelPipeline(BasePipeline):
+    
+    def _target_encoder(self):
+        pass
+
+    def feed_shape_type_def(self):
+        return (
+            (
+                {
+                    "tokens": tf.int32,
+                    "mask": tf.float32,
+                    "mlm_weights": tf.float32,
+                    "mlm_ids": tf.int32,
+                    "mlm_positions": tf.int32
+                },
+            ),
+            (
+                {
+                    "tokens": TS([self.config.max_length, 2]),
+                    "mask": TS([self.config.max_length]),
+                    "mlm_weights": TS([self.config.max_masked_tokens * self.config.batch_size]),
+                    "mlm_ids": TS([self.config.max_masked_tokens * self.config.batch_size]),
+                    "mlm_positions": TS([self.config.max_masked_tokens * self.config.batch_size]),
+                },
+            ),
+        )
+
+    def text_to_tokens_mask(self, X, Y=None):
```

Comment:
Here's one point of interaction between context and mlm. The other one currently is context and chunk_long_sequences (I hope to fix that soon, but ideally not block merging the current stuff for this?)

I added a quick line in the docs in my pr.